pytorch_optimizer/optimizer/soap.py

```diff
@@ -123,9 +123,11 @@ def get_orthogonal_matrix(mat: torch.Tensor) -> List[torch.Tensor]:
                 continue
 
             try:
-                _, q = torch.linalg.eigh(m + 1e-30 * torch.eye(m.shape[0], device=m.device))
+                _, q = torch.linalg.eigh(m + 1e-30 * torch.eye(m.shape[0], device=m.device, dtype=m.dtype))
             except Exception:  # pragma: no cover
-                _, q = torch.linalg.eigh(m.to(torch.float64) + 1e-30 * torch.eye(m.shape[0], device=m.device))
+                _, q = torch.linalg.eigh(
+                    m.to(torch.float64) + 1e-30 * torch.eye(m.shape[0], device=m.device, dtype=m.dtype)
```

Comment:
```suggestion
                    m.to(torch.float64) + 1e-30 * torch.eye(m.shape[0], device=m.device, dtype=torch.float64)
```

Pytorch would automatically handle this, adding different types of tensors, but, I think setting the dtype of torch.eye to `float64` would be good to align with the type (`m.to(torch.float64)`) instead of `m.dtype` in the matter of computation.